Sources/NIOCore/ChannelPipeline.swift

```diff
@@ -1503,7 +1512,7 @@ public final class ChannelHandlerContext: ChannelInvoker {
 
     /// Signal to the next `ChannelHandler` that a read burst has finished.
     public func fireChannelReadComplete() {
-        self.next?.invokeChannelReadComplete()
+        executeIfNotNil(&self.next) { $0.invokeChannelReadComplete() }
```

Comment:
Sadly I don't think we can make this optimisation: it'll violate the law of exclusivity. It's possible for us to synchronously reconfigure the pipeline, which will write to this property.

Reply:
Yeah, I was looking at something referenced in an old issue which suggested this is something we could try although I can see that when I apply this particular variant across some tests they explode with access issues.

I'm studying the other issue though and think I just overlooked something and there may be something we can do here.